### PR TITLE
Milestone3

### PR DESCRIPTION
**Pull request by abouhid**
In this PR we :
- Make the BooksForm component stateful
- Connect the REMOVE_BOOK action to the component

OBS: As you can see I did not follow through exactly what was asked on the milestone, as I wanted to learn how to work with hooks (useDispatch) and with the React-Forms package (onSubmit). I hope it is not a big deal, since the purpose of all of this is learning.

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
   return (
     <ul>
       <li>
-        Book ID:{book.id} - Book title:{book.title} - Book category:
+        Book ID: {book.id} - Book title: {book.title} - Book Category:{" "}
         {book.category}
+        <Button onClick={() => dispatch(removeBook(book))}>Delete Book</Button>
       </li>
     </ul>
   )
```
Comment (christianotieno) [inline src/components/Book.js]: ## Component Implementation 
* [x] This component should add `Remove book` button to the book row [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/RemoveBookButton.png)
* [x] It should be attaching passed `handleRemoveBook` to on click event of `Remove book` button
* [x] It should be staying as a presentational component

#### Kindly Implement these changes as they are mandatory, team!

File: src/containers/BooksList.js [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
   })
 
   return <div>{displayBooks}</div>
 }
 
-export default BooksList
+export default connect(state => ({ state }), { createBook, removeBook })(
+  BooksList
+)
+
```
Comment (christianotieno) [inline src/containers/BooksList.js]: ## Component Implementation 
* [x] This component should be implementing the `handleBookRemove` [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/ImplementsRemoveBook.png)
* [x] It should be connecting the `REMOVE_BOOK` action
* [x] It should also be passing the `handleRemoveBook` to Book component [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/ImplementsRemoveBook.png)

#### Kindly Implement these changes as they are mandatory, team!

**Review by christianotieno — CHANGES_REQUESTED** [review]
## Changes required ♻️ 
Hello @abouhid & @peterrobert 👋 
Congratulations on the work done in this milestone.
We do have some few changes to take care of before we proceed, down below 👇 

### Happy coding 💻 
#### Christian Otieno (CR)

File: src/containers/BooksList.js [hunk 3]
```diff
@@ -1,14 +1,28 @@
...
+  const handleRemoveBook = book => {
+    dispatch(removeBook(book))
+  }
+
+  const allBooks = store.getState().reducer
+
+  const displayBooks = allBooks.map(book => {
+    return (
+      <Book handleRemoveBook={handleRemoveBook} key={book.id} book={book} />
+    )
```
Comment (shloch) [inline src/containers/BooksList.js:20]: great job with the modifications

File: src/components/Book.js [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
 
-const Book = ({ book }) => {
+const Book = ({ book, handleRemoveBook }) => {
   return (
     <ul>
       <li>
-        Book ID:{book.id} - Book title:{book.title} - Book category:
+        Book ID: {book.id} - Book title: {book.title} - Book Category:{" "}
         {book.category}
+        <Button onClick={() => handleRemoveBook(book)}>Remove Book</Button>
```
Comment (shloch) [inline src/components/Book.js:10]: nice

**Review by shloch — APPROVED** [review]
## Hello @abouhid 

Great job with the ameliorations. You have incorporated all the modifications pointed out by the former TSE.
accomplished

Based on this, you get a thumbs up. Happy coding

**Pull request merged**